tests/test_real_pins.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import (
+    unicode_literals,
+    absolute_import,
+    print_function,
+    division,
+    )
+str = type('')
+
+
+import io
+import subprocess
+
+import pytest
+
+from gpiozero import PinFixedPull, PinInvalidPull, PinInvalidFunction
+
+
+# This module assumes you've wired the following GPIO pins together
+TEST_PIN = 22
+INPUT_PIN = 27
+
+
+# Skip the entire module if we're not on a Pi
+def is_a_pi():
+    with io.open('/proc/cpuinfo', 'r') as cpuinfo:
+        for line in cpuinfo:
+            if line.startswith('Hardware'):
+                hardware, colon, soc = line.strip().split(None, 2)
+                return soc in ('BCM2708', 'BCM2709', 'BCM2835', 'BCM2836')
+        else:
+            return False
+pytestmark = pytest.mark.skipif(not is_a_pi(), reason='tests cannot run on non-Pi platforms')
+del is_a_pi
+
+
+# Try and import as many pin libraries as possible
+PIN_CLASSES = []
+try:
+    from gpiozero.pins.rpigpio import RPiGPIOPin
+    PIN_CLASSES.append(RPiGPIOPin)
+except ImportError:
+    RPiGPIOPin = None
+try:
+    from gpiozero.pins.rpio import RPIOPin
+    PIN_CLASSES.append(RPIOPin)
+except ImportError:
+    RPIOPin = None
+try:
+    from gpiozero.pins.pigpiod import PiGPIOPin
+    PIN_CLASSES.append(PiGPIOPin)
+except ImportError:
+    PiGPIOPin = None
+try:
+    from gpiozero.pins.native import NativePin
+    PIN_CLASSES.append(NativePin)
+except ImportError:
+    NativePin = None
+
+
+@pytest.fixture(scope='module', params=PIN_CLASSES)
+def pin_class(request):
+    # pigpiod needs to be running for PiGPIOPin
+    if request.param.__name__ == 'PiGPIOPin':
+        subprocess.check_call(['sudo', 'pigpiod'])
```

Comment:
Yup, all good points; though I think I'm happy to leave off testing the remote capabilities of pigpiod for the time being (given that, when running locally it's still operating over a localhost connection, so if it works there it should all work remotely too). Maybe eventually I could imagine having a dual pi rig setup for testing remote stuff too, but that's way off yet.